Add support for .net Framework 4.8 and .net 5

ahmed-abdelrazek opened the pull request:
update all packages to latest 

add support for .net Framework 4.8 and .net 5 next to .net Framework 4.7.2 and .net core 3.1 for the main project

and replace .net core 3.1 with .net Framework 4.8 and .net 5 for the demo app

ahmed-abdelrazek committed:
Add support for .net Framework 4.8 and .net 5

thoemmi commented:
Thanks for reminding me to add .NET 5 support. However, I have to reject your other changes. Users of this library using .NET Core 3.1 wouldn't be able to update. Also I don't want to force everyone to update their Microsoft.Extensions.* packages to 5.0.

I have added `net5.0-windows` in 0b2ec4fcf721a028e29e527c38a02d0a437baa50.

ahmed-abdelrazek commented:
the library still support .net framework 4.7.2 and .net core 3.1 i just added .net 5 and .net framework 4.8 to it

i removed .net framework 4.7.2 and .net core 3.1 from the demo app only not the library

thoemmi commented:
Sorry, I should have read your changes more carefully. I didn't add *net48* in my commit because I didn't want to install the .NET 4.8 SDK :wink: And I don't think that adding *net48* would provide any benefits in addition to *net472*.

Anyway, thanks four your effort.
